src/backend/metal/src/lib.rs

```diff
@@ -210,6 +210,9 @@ impl Instance {
             return self.create_from_layer(view);
         }
 
+        // TODO: figure out why this is needed. Dolphin goes the `use_currect` path
+        // and crashes on exit when trying to release `NSView` otherwise.
+        msg_send![view, retain];
```

Comment:
This seems right to me (even for the `existing` path), so maybe the `TODO` isn't needed?

Unless I'm misunderstanding, both `create_from_layer` and `create_from_uiview` appear to always `retain` already because we'll always `release` during `SurfaceInner`'s `drop` if `self.view` is `Some` (which it appears to be here).

In other words:

- With this change, we unconditionally `retain` here, and `release` during `drop` (i.e. add `1` to the ref count here and subtract `1` during `drop`). This seems right.
- Without this change, we don't `retain` here for `existing`, yet still `release` during `drop` (i.e. add `0` to the ref count here and subtract `1` during `drop`). This doesn't seem right, because we might cause `view` to be released too early if something else was holding a reference.